client/src/observatory/results.tsx

```diff
@@ -0,0 +1,668 @@
+  return Object.keys(result.tests).length !== 0 ? (
+    <section className="tab-content">
+      <figure className="scroll-container">
+        <table className="tests">
+          <thead>
+            <tr>
+              <th>Test</th>
+              <th>Score</th>
+              <th>Reason</th>
+              <th>Recommendation</th>
+            </tr>
+          </thead>
+          <tbody>
+            {Object.entries(result.tests).map(([name, test]) => {
+              return (
+                <tr key={name}>
+                  <td data-header="Test">
+                    <Link href={test.link}>{test.title}</Link>
+                  </td>
+                  {test.pass === null ? (
+                    <td data-header="Score">-</td>
+                  ) : (
+                    <td className="score" data-header="Score">
+                      <span>
+                        <span className="obs-score-value">
+                          {fixMinusSymbol(`${test.score_modifier}`)}
+                        </span>
+                        <PassIcon pass={test.pass} />
+                      </span>
+                    </td>
+                  )}
+                  <td
+                    data-header="Reason"
+                    dangerouslySetInnerHTML={{
+                      __html: test.score_description,
+                    }}
+                  />
+                  <td
+                    data-header="Advice"
+                    dangerouslySetInnerHTML={{
+                      __html:
+                        test.recommendation || `<p class="obs-none">None</p>`,
+                    }}
+                  />
+                </tr>
+              );
+            })}
+          </tbody>
+        </table>
+      </figure>
+    </section>
+  ) : null;
+}
+
+function ObservatoryHistory({ result }: { result: ObservatoryResult }) {
+  return result.history.length ? (
+    <section className="tab-content">
+      <figure className="scroll-container">
+        <table className="history">
+          <thead>
+            <tr>
+              <th>Date</th>
+              <th>Score</th>
+              <th>Grade</th>
+            </tr>
+          </thead>
+          <tbody>
+            {[...result.history]
+              .reverse()
+              .map(({ scanned_at, score, grade }) => (
+                <tr key={scanned_at}>
+                  <td data-header="Date">
+                    {new Date(scanned_at).toLocaleString([], {
+                      dateStyle: "full",
+                      timeStyle: "medium",
+                    })}
+                  </td>
+                  <td data-header="Score">{score}</td>
+                  <td data-header="Grade">{fixMinusSymbol(grade)}</td>
+                </tr>
+              ))}
+          </tbody>
+        </table>
+      </figure>
+    </section>
+  ) : null;
+}
+
+function ObservatoryCookies({ result }: { result: ObservatoryResult }) {
```

Comment:
1. Consider extracting each of these tabs to a separate file, such as `./results/cookies.tsx`.
2. Since we only use `result.tests["cookies"]`, why not reduce the prop to that instead?

Reply:
1. Yes to separating those components into dedicated files - done.
2. I'd like to keep the tab component's properties consistent, having the `result` available is nice and there is no real performance benefit.